### DISCUSSION TITLE
unable to get bytemask working.

### DISCUSSION THREAD

**Discussion by AliAkhgar** [PatilShreyas/bytemask#14] [Q&A] [answered]
Hi, 
I've added bytemask to my android project but after exporting in both debug and release and opening in jadx, I can see that none of my strings are encrypted by bytemask.

Here is My gradle plugins on top of my gradle app level file.
```
plugins{
    id("com.android.application")
    id("org.jetbrains.kotlin.android")
    id('com.google.gms.google-services')
    id("com.facebook.react")
    id("dev.shreyaspatil.bytemask.plugin") version "1.0.0-alpha02"
}
```
 here is my bytemask config at the end of app level gradle.
```
bytemaskConfig {
    defaultPropertiesFileName.set("bytemask.properties")
    className.set("BytemaskConfig")
    configure("debug") {
        enableEncryption.set(true)

        encryptionKeySource.set(new KeySource.SigningConfig("debug"))
    }

    configure("release") {
        enableEncryption.set(true)

        encryptionKeySource.set(new KeySource.SigningConfig("release"))
    }
}
```

And here is my jadx screenshot.
<img width="994" alt="Screenshot 1403-04-04 at 9 45 11 at night" src="https://github.com/PatilShreyas/bytemask/assets/19927869/3da1533f-3ec9-4bf9-8c94-fc4f2d79c97f">

What am i missing out here, respectively?
gradle task :bytemaskDebug is being executed fine also.

**Comment by AliAkhgar** [comment]
i've read docs all over again, and to my understanding bytemask will encrypt properties inside bytemask.properties file and will expose those properties inside a class e.g. BytemaskConfig to be used safely in our programming logic; right or wrong?
i though it would just encrypt plain text strings inside java code.

**Comment by PatilShreyas** [comment]
@aliakhgar the screenshot you attached here, I'm just seeing MainApplication file. Can you check the content of `BytemaskConfig`?

So it just masks the strings present in .properties file into a separate class altogether.

**Comment by AliAkhgar** [comment]
Yes here is the screenshot of bytemask config.
so we should reference the keys exposed in bytemask in our programming logic right?
<img width="689" alt="Screenshot 1403-04-05 at 6 20 05 in the evening" src="https://github.com/PatilShreyas/bytemask/assets/19927869/a52d9f82-5e24-4fd0-b140-0d57f707f041">

<img width="689" alt="Screenshot 1403-04-05 at 6 18 09 in the evening" src="https://github.com/PatilShreyas/bytemask/assets/19927869/3b52a1c4-4d7c-4392-b508-d4ad244cbcb6">

**Comment by PatilShreyas** [comment, marked as answer]
Correct, you should be using them directly in your code. I also have demoed this in the docs with images. But if you check this, these strings are not plain and not easily reversible for the attacker. Also, once you apply R8 code shrinker, this becomes more trickier.

**Comment by AliAkhgar** [comment]
thanks for the info, i've been thinking of referencing these keys to js layer for react native which seems a bad idea now that i think of the purpose of this library.
Thanks for the info and very useful plugin.
feel free to close.
@PatilShreyas

**Comment by PatilShreyas** [comment]
Thanks. Feel free to discuss again.
